counsel.el

```diff
@@ -2063,6 +2063,65 @@ By default `counsel-bookmark' opens a dired buffer for directories."
    ("r" ,(counsel--apply-bookmark-fn #'counsel-find-file-as-root)
         "open as root")))
 
+;;** `counsel-bookmarked-directory'
+(defcustom counsel-bookmarked-directory-default-name "%s"
+  "Default bookmark name created by `counsel-bookmarked-directory-add."
+  :group 'ivy
+  :type 'string)
+
+(defun counsel-bookmarked-directory--path-separator ()
+  "Get the path separator for the current system."
+  (let* ((parent (expand-file-name "a"))
+         (i (length parent)))
+    (substring (expand-file-name "b" parent) i (1+ i))))
+
+(defconst counsel-bookmarked-directory-separator
+  (counsel-bookmarked-directory--path-separator))
```

Comment:
You should use the function `ivy--dirname-p` instead of this. Even on non-POSIX systems, Emacs understands POSIX filenames, so Elisp programs should almost never have to deal with path separators, which is very brittle; see [`(elisp) File Names`](https://www.gnu.org/software/emacs/manual/html_node/elisp/File-Names.html).